\documentclass[a4paper,12pts]{report}

\title{Thesis Proposal}
\author{Kevin Esoh}
\date{\today}

\begin{document}
\section{Background}
Severe \textit{Plasmodium falciparum} malaria claims the lives of thousands yearly especially children below 5 years old.  \textit{P. falciparum} is the most virulent malaria parasite strain and the most prevalent in  Africa, accounting for 99\% of all cases[1]. Low hemoglobin count, parasite sequestration in deep microvasculature and occlusion of blood vessels leading to coma and consequently death are common features of the debilitating disease[2, 3]. Whilst only a small proportion (about 1-2\%) of malaria cases progresses to severe malaria, most cases remain asymptomatic or uncomplicated[4]. In fact,  in some hyper-endemic areas, a sub-set of cases rarely develop into severe malaria. Epidemiologic studies suggest genetic factors are involved in these phenotypic differences[5]. However, most polymorphisms  (mainly single nucleotide polymorphisms - SNPs) that have been found in association with complex disease explain only a little percentage of the total phenotypic variation[6]. Hence, many more genetic variations may exist, contributing small effects to the total penotypic variation.

The protective role of several erythrocyte, immune system components, regulatory elements and cell adhesion molecule polymorphisms against various phenotypes of malaria has been well established. Hemoglobin S (HbS-the sickle cell trait) involves one of the most characterized polymorphisms with fixed protective effect against severe malaria across Sub-Saharan Africa[7]. Like Hemoglobin C and blood type O of the ABO blood group system, HbS infected erythrocytes have been linked to reduced cytoadherence[8, 9]. The G6PD deficiency is a classical example of a condition with multiple allelic variants and heterogeneous effects on clinical malaria phenotypes[10, 11, 12]. Whilst found in association with decreased risk of cerebral malaria (CM) across Africa, it is also associated with increased risk of severe malaria anemia (SMA)[11]. Several polymorphisms have been found on the G6PD gene in Africa [12, 13, 14, 15]. SNPs affecting cell surface receptors have received particular attention due to their involvement in the malaria disease process as these receptors are required by the parasite for their gliding motility[12], invasion of host cells[13, 14] and sequestration/rosseting by cytoadherence[8, 9, 15]. However, findings have been contradictory. For example, the ICAM-1 gene which was shown to associate with severe malaria malaria in 223 children in South-West Nigeria[16] and 547 Kenyan subjects (ICAM-1Kilifi)[17] showed no association in studies conducted in case-control subjects of the Gambia [18], Senegal[19] and Gabon[20]. ICAM-1(CD54) interacts with PfEMP1 and is for cytoadherence. 

The interaction of P. falciparum with the immune system determines disease outcomes and many genes have been shown to be involved. Immune response is partially mediated by immunoglobulin G following release of hemozoin and other proinflammatory molecules like glycosylphosphatidylinositol (GPI) involving toll-like receptors (TLR). Variants in the TLR4 gene have been shown to increase some infectious disease risk [22]. SNPs in the histidine-Arginine variant (H/R131) of the Fc gamma receptor IIa (FCGR2A or CD32) of  IgG that influence binding activity increases SMA susceptibility in Ghanaian children[23]. The same trend is found in West Africa[24], the Fulani and Masaleit ethnic groups of eastern Sudan[25] and in western Kenya[26]. Similarly, several polymorphisms in the major histocompartibility complex (human leucocyte antigen - HLA) gene regions have been shown to either increase or decrease susceptibility to severe malaria in African populations [Osafo-Addo, 2008]. The MHC genes are the most polymorphic in vertebrates and the origin of this repertoir of diversity remains open to question [Garamszegi, 2014]. SNPs in the promoter region of Tumor necrosis factor (TNF) that influence its plasma concentration have been implicated in the malaria disease process. Notably, the -308AA genotype which has been highly characterized in many conditions is associated with decreased malaria rates in young Tanzanian children while the opposite observation is made with the -1031CC genotype in the same population[Gichochi-Wainaina]. Considerable heterogeneity is seen in the association of HLA alleles with the malaria disease[Garamszegi, 2014]. The regulation of the balance between anti- and pro-inflammatory cytokines is important in determining the severity of malaria. SNPs in IL-1beta locus are implicated in the susceptibility to uncomplicated malaria in Ghanaian children[Gyan, 2002]. Polymorphisms in the promoter position -954 of the nitric oxide synthase 2 (NOS2) gene showed protective role in Uganada [Lwanira, 2017] and Ghanaian children with severe malaria [Dzodzomenya, 2018, Cramer, 2004] while polymorphism at -1173 increased risk of disease [Cramer, 2004]. Inducible NOS (iNOS) is expressed in hepatocytes during pathological conditions[Iwakiri,2015]. It is possible that the NOS gene evolved to upregulate its expression in response to P. falciparum. However, some findings have been contradictory. No association with malaria severity was previously seen for the NOS haplotypes in Tanzania [Levesque, 1999], but a novel promoter SNP at -1173CT was shown to associated with protection against uncomplicated (UM) malaria in Tanzanian children with protective effect against SMA in Kenyan children[Hobbs, 2002].

Candidate/targeted gene analyses of case-control and family-based trios studies have been vital in unraveling these mutations. Genome-wide association studies (GWAS) have enabled the discovery of thousands of disease-related SNPs within a short period of time, and at a lower cost. The first  linkage disequlibrium (LD)-based GWAS of malaria in Africa carried out on a sample of Gambian children with severe malaria found considerable population stratification and significant attenuation of association signals (Jallow et al., 2009) and a multipoint imputation replicated a strong signal of association in the  HbS locus (Jallow et al., 2009). Several polymorphisms have been found in association with severe malaria across Africa including the ATP2B4 locus in Ghana(Timmann et al. 2012), HBB, ABO and ATP2A loci[MalariaGEN, 2015], the FREM3, MARVELD3 and the glycophorin gene clusters (GYPE, GYPB and GYPA) on chromosome 4 of Ghanaian, the Gambian, Kenyan and Malawian subjects  (Timmann et al. 2012; MalariaGEN, 2015) and the interleukin receptor genes IL-23R and IL-12RBR2, and the kelch-like protein KLHL3 gene in Tanzania(Ravenhall et al., 2018). A weak LD and an extensive genetic diversity between and within African populations[Busby, 2016] are the two inherent methodological challenges to the success of GWAS in Africa[Teo, 2010; ]. The lack of a representative reference haplotype panel fit for imputation of all African populations and the high burden of infectious diseases further impose significant challenges to African GWAS[Teo, 2010]. These findings support the involvement of host genetics in the determination of malaria disease susceptibility and show the complexity of these genetic factors, hence suggesting the existing of possibly more variations that are yet to be discovered.

The story has not been different in the case of Cameroon where candidate gene case-control studies have found several polymorphisms in association with malaria and severe malaria. The well characterized SNPs in the HbS, HBB and ABO (which protects against hyperparasitaemia) loci have been replicated. The protective effect of SNPs in the NOS2, IL-10, IL-17RE and ADCY9 genes have shown, while SNPs in the EMR1, RTN3, IL-10 rs3024500 AA, hHbS rs334 TT and IL17RD rs6780995 GA have been found to increase susceptibility to various malaria disease outcomes [Apinjoh 2013, 2014]. The G6PD deficiency has also been observed. However, no whole-genome GWAS has been performed to capture the broad spectrum of variations that may have occurred  in the entire genome of Cameroonian malaria-endemic populations. Here, we use a whole-genome GWAS approach of 6.. case and 6.. control subjects from three malaria-endemic regions and three ethnic groups of Cameroon in a bid to uncover novel variations that may not have been captured by previous studies, but however play important roles in the clinical outcome of malaria. Uncovering these variations will be invaluable in gaining insight into the molecular basis of malaria disease process, key pathogenesis mechanisms, protective immunity and host-parasite interactions. This will in turn be important in informing better vaccine development strategies.

\end{document}